REFERENCE.md

```diff
@@ -172,202 +172,268 @@ Manage the firewall rules for the Samba services.
 
 ##### <a name="-samba--workgroup"></a>`workgroup`
 
-Data type: `Optional[String]`
+Data type: `Variant[Undef, String]`
```

Comment:
What is reason for changing `Optional[String]` to Variant[Undef, String]`? Is this to force explicit definition of all these parameters?